### PR TITLE
306 documentation improvements

### LINKED ISSUE #306: Documentation improvements
- "Index" section. The README is too long, we need links to the subsections near the top
- The `epinio push` of the sample app doesn't explain how to get the sample-app (`git clone` the repo? include release assets? new repo?)
- What drives Epinio's new features? What comes next? Why is there no far future Vision? (clearly state what Epinio is)

### PR DESCRIPTION
**Pull request by jimmykarily**
Fixes #306

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,13 +141,25 @@ application to your Kubernetes cluster with Epinio.
 NOTE: If you want to know the details of the `epinio push` process, read this
 page: [detailed push docs](/docs/detailed-push-process.md)
 
-Run the following command for any supported application directory. If you just
-want an application that works use the one inside the [sample-app directory](assets/sample-app).
+If you just want an application that works use the one inside the
+[sample-app directory](assets/sample-app). You can copy it on your system with
```
Comment (andreas-kupries) [inline README.md:145]: `on your system` -> `to your system`

File: README.md [hunk 2]
```diff
@@ -99,13 +141,25 @@ application to your Kubernetes cluster with Epinio.
...
+```
+
+To push the application run the following command:
 
 ```bash
 $ epinio push sample sample-app
 ```
 
+where `sample` is the name you want to give to your application. This name has
+to be unique within Epinio. `sample-app` is path to the directory where your
```
Comment (andreas-kupries) [inline README.md:160]: `within ... the targeted organization in Epinio`.

You can have multiple `sample` applications, if they are in different orgs.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jimmykarily) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (jimmykarily) [reply README.md]: done.

**Review by kkaempf — APPROVED** [review]
lgtm

**Review by agracey — APPROVED** [review]
LGTM

**Comment by jimmykarily** [comment]
Tests are flaking. Nothing to fix on this PR though (just README changes).

**Pull request merged**